src/main/scala/com/mozilla/telemetry/pings/MobileEvent.scala

```diff
@@ -0,0 +1,97 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+package com.mozilla.telemetry.pings
+
+import com.mozilla.telemetry.heka.Message
+import com.mozilla.telemetry.pings.MobileEventPing.OptionToBoolean
+import org.json4s.JsonDSL._
+import org.json4s._
+
+case class MobileEventPing(arch: String,
+                           clientId: String,
+                           created: Long,
+                           device: String,
+                           events: Seq[Event],
+                           locale: String,
+                           os: String,
+                           osversion: String,
+                           seq: Integer,
+                           settings: MobileSettings,
+                           v: String,
+                           meta: Meta) extends SendsToAmplitude {
+
+  override def getClientId: Option[String] = Some(clientId)
+
+  def sessionStart: Long = created
+
+  def getSessionId: Option[String] = Some((events.map(_.timestamp).max).toString)
+
+  def getOsName: Option[String] = Some(os)
+
+  def getOsVersion: Option[String] = Some(osversion)
+
+  def getCreated: Option[Long] = Some(created)
+
+  override def pingAmplitudeProperties: JObject = {
+    ("device_id" -> getClientId) ~
+    ("device_model" -> device) ~
+    ("arch" -> arch) ~
+    ("locale" -> locale) ~
+    ("user_properties" ->
+      ("pref_defaultSearchEngine" -> settings.defaultSearchEngine) ~
```

Comment:
Just want to point out that if it's easier for the PMs, you could make these pretty-printed names instead (e.g. `Default-Search-Engine` or similar). I'm not sure if spaces are allowed.

Reply:
Hey @justindarc - would you like some nice, normalized property names here [1]?
I copied these 1:1 from your telemetry data as there seem to be couple of naming conventions in use, but maybe pretty-printing them as Frank suggests would make your life easier?

[1] https://github.com/mozilla/telemetry-streaming/blob/ios-amplitude/src/main/scala/com/mozilla/telemetry/pings/MobileEvent.scala#L42-L59

Reply:
Yes, if you could map them to pretty-printed labels, that would probably be easier for the PMs. Do you need me to provide you the labels or can you infer them from what's there?